### ISSUE TITLE
Want the ability to see all quotes of a post

### ISSUE THREAD

**Issue by alexgleason** [nostr-protocol/nips#616]
I was reluctant to bring this up because of all the drama of NIP-10 and NIP-27, but I think the current solution is still wrong. Quotes of a post should be indexable and able to be viewed.

This is related to #523 

![IMG_20230620_091944](https://github.com/nostr-protocol/nips/assets/3639540/35e01199-3524-45f6-baf2-886eaf514af6)

I would propose using a "q" tag. It's effectively the same as "e" except it's explicitly a quoted event on kind 1 notes, where "e" is just a reply.

**Comment by fiatjaf** [comment]
Are you a nym of @arthurfranca?

**Comment by arthurfranca** [comment]
Heheh I've also pointed out in the past that using `e` tags to reference events in any context has problems (NIP-10 marks can't be filtered relay-side).

However, when i brought back to life NIP-18 and also when I tried to change how reply threads should be built, I had to accept the "rule" that whatever main nostr clients were doing would become the NIPs, even if it was a poor implementaion. So if I want a different behavior, I must send a PR **before** people had already implemented things.

So the immutable reality is this: one variant of the reposts is kind 6 (this you can filter) and the other one is kind 1 with mention and e tag (this you can't really filter).

For the kind 1 variant, we could add a `["q", "true"]` tag (NIP-18 Quote Reposts) so to be able to filter this kind of repost. It would be a good change that would make no harm imo as it would be just an addition that wouldn't break things. What do you think?

But good luck convincing Damus, Amethyst, Coracle, Snort, Iris, Gossip etc to add it or waiting for them to read the updated NIP.

**Comment by arthurfranca** [comment]
@alexgleason also check [this NIP-17 PR](https://github.com/nostr-protocol/nips/pull/605) which would also help.

**Comment by fiatjaf** [comment]
You know Twitter didn't display quote counts at all until last year perhaps? I think they had a data model pretty similar to Nostr.

**Comment by alexgleason** [comment]
I think it's a feature users expect, but I definitely see it as part of a long-term plan. For now I just want an issue open.

I'll probably try a "q" tag in Ditto later and see how that goes. As arthurfranca said, it's possible to do this without breaking backwards-compatibility.

**Comment by arthurfranca** [comment]
`["q", "true"]`, `["q", "t"]` or `["q", "1"]` which one is best?

**Comment by alexgleason** [comment]
@arthurfranca It would have to be `["q" <event-id>]` so you could query the quotes of a post. For backwards compat, clients could still add an "e" tag.

**Comment by arthurfranca** [comment]
Ok, just saw a random thread on nostr saying Damus (Highlighter also) is using `["q" <event-id>]` for atleast 6 weeks already

**Comment by ryzizub** [comment]
Yes, the 'q' tag is a good way to handle quotes, I like it a lot. I think we should also keep the e tag with mention marker for a while as fallback but at the end we don't need that or i'm wrong (When I first saw Damus q tag i thought we need both, but i don't find reason keeping that mention e tag, root and reply are still needed for sure)

**Comment by fiatjaf** [comment]
Using a `q` tag doesn't sound bad.

@v0l @jb55 @staab @pablof7z what do you have to say about this?

**Comment by pablof7z** [comment]
yeah, I like `q` tag a lot and afaik damus already uses it and so does highlighter; I think that's the right way forward

no `e` tag though; that's redundant and quotes don't belong in replies anyway so it's not breaking prior implementations, it's actually fixing them

**Comment by fiatjaf** [comment]
I like quotes in replies.

**Comment by jb55** [comment]
yeah damus already uses `q`

**Comment by jb55** [comment]
> I like quotes in replies, but I guess it is fine to not have them.

if you're replying then it's already quoted above the thing you're replying to in some sense, so not sure when this makes sense. many clients already use an e tag when quoting and it shows up in damus threads, looks a bit weird.

**Comment by arthurfranca** [comment]
@jb55 you created kind 6 reposts that have the stringified JSON of the reposted event inside the `.content` for fast lookups (no need to ask relay for the event using the `e` tag value).

But for quotes you decided not to add the stringified JSON in some tag. Why is that?

**Comment by jb55** [comment]
it was always just an optimization.. I still could? damus got better at fetching notes inline. it's still not as reliable as kind6 reposts but it works well enough.

**Comment by jb55** [comment]
now that we have longer notes like longform, the optimization is looking less attractive.

**Comment by staab** [comment]
q makes sense to me
